File: sdk/sdk-api-php/utils/SignUtil.php
<?php
namespace xpay;

use ReflectionClass;
use ReflectionProperty;

/**
 * 签名、验签工具类
 * Class SignUtil
 * @package utils
 */
class SignUtil {
    const MS5_BOUND_SYMBOL = "&key=";

    /**
     * 签名
     * @param string $signData 需要签名的数据
     * @param string $signType 签名类型
     * @param string $priKey 用以签名的私钥
     * @return string
     * @throws \exceptions\SDKException
     * @throws SDKException
     */
    public static function sign(string $signData, string $signType, string $priKey): string
    {
        if("2" === $signType){
            return RSAUtil::sign($signData, $priKey);
        }else{
            throw new SDKException(SDKException::BIZ_ERROR, "未支持的签名类型：" . $signType);
        }
    }

    /**
     * 验签
     * @param string $signData 需要验签的数据
     * @param string $signParam 需要被校验签名的源数据
     * @param string $signType 签名类型
     * @param string $pubKey 用以验签的公钥
     * @return bool
     * @throws \exceptions\SDKException
     * @throws SDKException
     */
    public static function verify(string $signData, string $signParam, string $signType, string $pubKey): bool
    {
        if("2" === $signType){
            return RSAUtil::verify($signData, $signParam, $pubKey);
        }else{
            throw new SDKException(SDKException::BIZ_ERROR, "未支持的签名类型：" . $signType);
        }
    }
}

File: sdk/sdk-api-java/src/main/java/com/xpay/sdk/api/utils/SignUtil.java
package com.xpay.sdk.api.utils;

import com.xpay.sdk.api.enums.SignType;
import com.xpay.sdk.api.exceptions.SDKException;

import java.nio.charset.StandardCharsets;

/**
 * 签名、验签的工具类
 * @author chenyf
 * @date 2018-12-15
 */
public class SignUtil {

    /**
     * 生成签名
     * @param data
     * @param key
     * @return
     */
    public static String sign(byte[] data, String singType, String key){
        if (SignType.RSA.getValue().equals(singType)) {
            return RSAUtil.sign(data, key);
        } else if (SignType.MD5.getValue().equals(singType)) {
            return genMD5Sign(data, key);
        } else {
            //抛出签名失败的异常
            throw new SDKException("签名失败，未预期的签名类型：" + singType);
        }
    }

    /**
     * 同步响应时验证签名
     * @param data
     * @param key
     * @return
     */
    public static boolean verify(byte[] data, String signature, String signType, String key){
        if(SignType.RSA.getValue().equals(signType)){
            return RSAUtil.verify(data, key, signature);
        }else if(SignType.MD5.getValue().equals(signType)){
            String signStr = genMD5Sign(data, key);
            return signStr.equals(signature);
        }else{
            return false;
        }
    }

    private static String genMD5Sign(byte[] signData, String key) {
        String dataStr = new String(signData, StandardCharsets.UTF_8);
        return CodeUtil.base64Encode(MD5Util.getMD5(dataStr + key));
    }
}
